src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -215,7 +220,7 @@ def normalize_disk_info(image_data_disks=None,
             info[i]['managedDisk'] = {'id': d}
             if data_disk_delete_option:
                 info[i]['deleteOption'] = data_disk_delete_option if isinstance(data_disk_delete_option, str) \
-                    else data_disk_delete_option.get(info[i]['name'], None)
+                    else data_disk_delete_option.get(d.split('/')[-1].split('.')[0], None)
```

Comment:
Why do we need to modify this logic?

Reply:
The previous code is incorrect but was not covered by the test.